### PR TITLE
#6886: Configurable CSW filters

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR adds possibility to configure CSW filters from localConfig and from advanced settings of Catalog

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Enhancement

## Issue

**What is the current behavior?**
#6886  &  #7084 

**What is the new behavior?**
The CSW filter is configurable as below
- `staticFilter `- filter is always applied, even when search text is NOT PRESENT
- `dynamicFilter `- filter is used when search text is PRESENT and is applied in `AND` with `staticFilter`. The template is used with ${searchText} placeholder to append search string

_Example_
```
{ 
  filter: {
    staticFilter: "<ogc:Or><ogc:PropertyIsEqualTo><ogc:PropertyName>dc:type</ogc:PropertyName><ogc:Literal>dataset</ogc:Literal></ogc:PropertyIsEqualTo></ogc:Or>",
    dynamicFilter: "<ogc:PropertyIsLike wildCard='*' singleChar='_' escapeChar='\\'><ogc:PropertyName>csw:AnyText</ogc:PropertyName><ogc:Literal>${searchText}*</ogc:Literal></ogc:PropertyIsLike>"
  }
}
```
Note: `filter` - For both static and dynamic filter, input only xml element contained within <ogc:Filter> (i.e. Do not enclose the filter value in <ogc:Filter>)<br>

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/translations/data.it-IT.json [hunk 1]
```diff
@@ -1422,6 +1422,18 @@
...
+            },
+            "filter": {
+                "static": {
+                    "label": "Filtro statico",
+                    "info": "Il filtro viene applicato sempre, anche nella ricerca vuota"
+                },
+                "dynamic": {
+                    "label": "Filtro dinamico",
+                    "info": "Il filtro viene utilizzato quando è presente il testo di ricerca e viene applicato in 'AND' con filtro statico",
+                    "helpText": "Usa il modello con il segnaposto ${searchText} per acquisire la stringa di ricerca"
```
Comment (offtherailz) [inline web/client/translations/data.it-IT.json:1434]:
```suggestion
                    "helpText": "Usa il segnaposto ${searchText} nel modello per inserire la stringa di ricerca"
```

**Review by offtherailz — CHANGES_REQUESTED** [review]
Works well. Only one fix:
 - [ ] In dashboard, the advanced settings for catalog are shown, but the filter is not used.

https://user-images.githubusercontent.com/1279510/133051946-10206942-a104-4627-9c39-e724ec7b2da0.mp4

**Review by offtherailz — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo please test this in DEV and let us know if we can backport to 2021.02.xx

**Comment by ElenaGallo** [comment]
Test passed, @dsuren1 please backport to 2021.02.xx

**Pull request merged**